crates/c-api/include/wasmtime/component/linker.h

```diff
@@ -0,0 +1,95 @@
+#ifndef WASMTIME_COMPONENT_LINKER_H
+#define WASMTIME_COMPONENT_LINKER_H
+
+#ifdef WASMTIME_FEATURE_COMPONENT_MODEL
+
+#include <wasm.h>
+#include <wasmtime/component/component.h>
+#include <wasmtime/component/instance.h>
+#include <wasmtime/error.h>
+#include <wasmtime/store.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct wasmtime_component_linker_t wasmtime_component_linker_t;
+
+typedef struct wasmtime_component_linker_instance_t
+    wasmtime_component_linker_instance_t;
```

Comment:
Mind deferring this type to later? I think there's going to be some tricky ownership questions we'll want to grapple with for this (e.g. handling the lifetime parameter in the Rust side of things)